heron/executor/src/python/heron_executor.py

```diff
@@ -107,61 +98,53 @@ def atomic_write_file(path, content):
   os.rename(tmp_file, path)
 
 def log_pid_for_process(process_name, pid):
-  """ log pid for process """
   filename = get_process_pid_filename(process_name)
   do_print('Logging pid %d to file %s' %(pid, filename))
   atomic_write_file(filename, str(pid))
 
 # pylint: disable=too-many-instance-attributes
 class HeronExecutor(object):
-  """ Heron executor """
+  """ Heron executor is a class that is responsible for running each of the process on a given
+  container. Based on the container id and the instance distribution, it determines if the container
+  is a master node or a worker node and it starts processes accordingly."""
   def __init__(self, args):
     self.max_runs = 100
     self.interval_between_runs = 10
     self.shard = int(args[1])
     self.topology_name = args[2]
     self.topology_id = args[3]
     self.topology_defn_file = args[4]
-    # args[5] is the instance distribution encoding
```

Comment:
See line 170 below. args[5] gets passed to a method that parses it and sets each of these:
```
self.instance_distribution = {}
self.stmgr_ids = []
self.metricsmgr_ids = []
self.heron_shell_ids = []
```

Once we add support for the state manager listener, we'd instead register the listener to call that method.